src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -315,6 +319,21 @@ private void EmitExpressionCore(BoundExpression expression, bool used)
             }
         }
 
+        private void EmitThrowExpression(BoundThrowExpression node, bool used)
+        {
+            this.EmitExpression(node.Expression, true);
+            var thrownType = node.Expression.Type;
+            if (thrownType?.TypeKind == TypeKind.TypeParameter)
```

Comment:
Verification requires that argument of throw is a reference type and does not consider constraints for that purpose. Box will be a no-op here. 
Consider it as a "cast to a reference type".